flash/core/data/data_source.py

```diff
@@ -543,3 +554,256 @@ def _get_classes(self, data):
             classes = data.distinct(label_path)
 
         return classes
+
+
+class LabelStudioDataset(Dataset):
+    """Dataset wrapping label studio annotations.
+
+    Each sample will be retrieved by checking result field of the annotation.
+
+    Args:
+        *js: json of export file
+        *img_folder: path to image folder of label studio
+    """
+
+    def __init__(self, js, img_folder, val=False):
+        self._raw_data = js
+        self._img_folder = img_folder
+        self.results = []
+        self.classes = set()
+        # iterate through all tasks in exported data
+        for task in self._raw_data:
+            for annotation in task["annotations"]:
+                # Adding ground_truth annotation to separate dataset
+                result = annotation["result"]
+                for res in result:
+                    t = res["type"]
+                    for label in res["value"][t]:
+                        # check if labeling result is a list of labels
+                        if isinstance(label, list):
+                            for sublabel in label:
+                                self.classes.add(sublabel)
+                                temp = {}
+                                temp["file_upload"] = task["file_upload"]
+                                temp["label"] = sublabel
+                                if annotation["ground_truth"] & val:
+                                    self.results.append(temp)
+                                elif not annotation["ground_truth"] or not val:
+                                    self.results.append(temp)
+                        else:
+                            self.classes.add(label)
+                            temp = {}
+                            temp["file_upload"] = task["file_upload"]
+                            temp["label"] = label
+                            if annotation["ground_truth"] & val:
+                                self.results.append(temp)
+                            elif not annotation["ground_truth"] or not val:
+                                self.results.append(temp)
+        self.num_classes = len(self.classes)
+
+    def __getitem__(self, idx):
+        r = self.results[idx]
+        # extracting path to file
+        p = os.path.join(self._img_folder, r["file_upload"])
+        # loading image
+        sample = default_loader(p)
+        # casting to list and sorting classes
+        sorted_labels = sorted(list(self.classes))
+        # checking index of class
+        label = sorted_labels.index(r["label"])
+        result = {DefaultDataKeys.INPUT: sample, DefaultDataKeys.TARGET: label}
+        return result
+
+    def __len__(self):
+        return len(self.results)
+
+
+class LabelStudioDataSource(DataSource):
```

Comment:
I think it would be cleaner to split this into seperate data sources for each modality (they could inherit from a base if there is some shared code). E.g. you would have:
- `LabelStudioImageDataSource`
- `LabelStudioTextDataSource`
- `LabelStudioVideoDataSource`

Reply:
Actually, I would make it explicit they are classification related data sources.

`LabelStudioImageClassificationDataSource`.

Reply:
Fixed